R/teal_data-extract.R

```diff
@@ -0,0 +1,38 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `teal_data` environment and limit the code to the necessary needed to build limited
+#' objects.
+#'
+#' @param names (`character`) names of objects included in `teal_subset` to subset
+#' @param x (`teal_data`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' data <- teal_data()
+#' data <- eval_code(data, "a <- 1;b<-2")
+#' data["a"]
+#' data[c("a", "b")]
+#'
+#' join_keys(data) <- join_keys(join_key("a", "b", "x"))
+#' join_keys(data["a"]) # should show empty keys
+#' join_keys(data["b"])
+#' join_keys(data)["a"] # should show empty keys
+#' join_keys(data)["b"]
+#'
+#' @rdname teal_data
+#'
+#' @export
+`[.teal_data` <- function(x, names) {
+  checkmate::assert_class(names, "character")
+  names_in_env <- intersect(names, ls(get_env(x)))
+  if (!length(names_in_env)) {
+    return(teal_data())
+  }
+
+  subset_qenv <- utils::getFromNamespace("[.qenv", "teal.code")
```

Comment:
Yeah, I was trying to specifically call method for the parent class :) will give it a shot with `NextMethod()`

Reply:
@llrs-roche I had issues calling `teal.code::[.qenv` as this has ticks in the names. It looks like you can call it like below

```r
teal.code:::`[.qenv`
```
but with `:::` 3 colons (even though this is exported from `teal.code`). 
That's why I decided to use `utils::getFromNamespace` so that we do omit a warning in R CMD CHECK that claims you rather not use `:::` in your codebase.

-----

Check out how it looks like with `NextMehod`. I needed to define `names` in the environment of the calling function, and it is not passed to `NextMethod`. This looks a bit weird IMHO. fd4fe27d6757d457987a7d5bd5d4e2ab3dfa266e

Maybe I just don't understand how `NextMethod` works but this is less clear IMHO than 

```r
subset_qenv <- utils::getFromNamespace("[.qenv", "teal.code")
```